### ISSUE TITLE
pylint custom configuration overrides plugin defaults settings.

### ISSUE THREAD

**Issue by matusvalo** [PyCQA/pylint#2635]
pylint overrides default configuration set by plugins using parameters set by rc file or command line arguments:
https://github.com/PyCQA/pylint/blob/fcc01516ae176ad3fdedc4497328105f3314e376/pylint/lint.py#L1568-L1571

pylint-django extends good-names in `register()` method [1]:
```python
    name_checker = get_checker(linter, NameChecker)
    name_checker.config.good_names += ('qs', 'urlpatterns', 'register', 'app_name', 'handler500')
    # we don't care about South migrations
    linter.config.black_list += ('migrations', 'south_migrations')
```

Currently plugin is not able to avoid overriding of configuration. pylint should provide a hook for plugins which is executed *after* configuration is set. Currently there are multiple open issues in pylint-django project affected by this defect: 
* https://github.com/PyCQA/pylint-django/issues/181 
* https://github.com/PyCQA/pylint-django/issues/167

### Steps to reproduce
1. Install django and pylint-django using pip
2. Create empty project: `django-admin startproject myproject`
3. Execute pylint with pylint-django plugin and with default configuration:
```
pylint --load-plugins=pylint_django myproject/myproject/urls.py

------------------------------------
Your code has been rated at 10.00/10
```
4. Execute pylint with django-pylint and custom configuration:
```
pylint --load-plugins=pylint_django --good-names=foo,bar  myproject/myproject/urls.py

************* Module myproject.urls
myproject/myproject/urls.py:19:0: C0103: Constant name "urlpatterns" doesn't conform to UPPER_CASE naming style (invalid-name)

-------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 10.00/10, -3.33)
```
### Current behavior
Currently, configuration provided by pylint-django is overriden by custom configuration (see [1] pylint-django configuration)

### Expected behavior
Custom configuration should not override pylint-django customization

### pylint --version output
pylint 2.1.1
astroid 2.0.4
Python 3.6.4 (default, Jan  7 2018, 15:53:53)
[GCC 6.4.0]

[1] https://github.com/PyCQA/pylint-django/blob/c5b5bfdef66453575074b36018ee716411bcb0a4/pylint_django/plugin.py#L18-L22

**Comment by PCManticore** [comment]
This makes sense, thanks for the issue!
